gradle/javadoc.gradle

```diff
@@ -27,8 +27,15 @@ task docs {
 def buildDocsUrl(String version, String packageName) {
     def docVersion = version.split('-').head()
     def versionList = docVersion.split('\\.')
-    def patch = (versionList[1] as int) - 1
-    docVersion = patch >= 0 ? "${versionList[0]}.${patch}" : '4.0'
+    def major = (versionList[0] as int)
+    def minor = (versionList[1] as int)
+
+    docVersion = "${major}.${minor}"
+    if (version.contains("-SNAPSHOT")) {
+        // If a snapshot set the version to the previous released version
+        docVersion = minor > 0 ? "${major}.${minor - 1}" : '4.0'
```

Comment:
If the version is `4.1.1-SNAPSHOT`, docVersion is `4.0`.  But should it be `4.1` instead?

Reply:
Its more correct but as we generally don't release any docs for snapshot versions all the urls will be broken. I thought it better to point to actually released docs so the docs are usable.

Reply:
Ah, but for the 4.1.1-SNAPSHOT release, we _have_ already released the 4.1 docs, so can't we point to them?

Reply:
Good catch